### DISCUSSION TITLE
Convert event based callbacks to use lambdas

### DISCUSSION THREAD

**Discussion by vurvdev** [wiremod/wire#2884] [Suggestions]
Certain events would be better represented with callbacks.

I did try to avoid having these events that would be better handled with lambdas, which is why there was never a timer or interval event, but the http and file events were deemed necessary enough to have an event counterpart.

Of course events that make sense globally like `tick`, `chat` etc will remain as events.

- [ ] http
- [ ] file

**Comment by Denneisk** [comment]
`playerMove`? Lol.

**Reply by Denneisk** [reply]
Also now this makes me feel bad for eventifying file even further, but no worries.
